src/app/keyboard/services/physical-keyboard-layout.service.ts

```diff
@@ -0,0 +1,129 @@
+import { KeyboardLayoutKey, KeyboardLayout } from '../value-objects/keyboard-layout'
+
+export class PhysicalKeyboardLayoutService {
```

Comment:
Lots of magic strings in here.
But in this context I think it make sense to keep them as is ;) 

Sidenote: I like the use of a matrix to represent the keyboard :)